app/src/main/java/io/islnd/android/islnd/messaging/ServerTime.java

```diff
@@ -0,0 +1,48 @@
+package io.islnd.android.islnd.messaging;
+
+import android.content.Context;
+import android.util.Log;
+
+import java.io.IOException;
+
+public class ServerTime {
+    private static final String TAG = ServerTime.class.getSimpleName();
+    private static final int REPETITIONS = 2;
+
+    private static long serverOffsetMillis = 0;
+    private static boolean inSync = false;
+    public static boolean isInSync() {
+        return inSync;
+    }
+
+    public static long synchronize(Context context) throws IOException {
+        serverOffsetMillis = MessageLayer.getServerTimeOffsetMillis(context, REPETITIONS);
+        inSync = true;
+        return serverOffsetMillis;
+    }
+
+    public static void synchronizeManually(long newServerOffsetMillis) {
+        serverOffsetMillis = newServerOffsetMillis;
+        inSync = true;
+    }
+
+    public static long toServerTimeMillis(long localTimeMillis) {
```

Comment:
I think this method can be private. Thinking about the interface... do we want this to be some utility class that is converting between different representations of time, or do we want to hide that complexity and expose a single notion of server time?